include/s3select.h

```diff
@@ -619,7 +636,11 @@ struct s3select : public bsc::grammar<s3select>
     {
       ///// s3select syntax rules and actions for building AST
 
-      select_expr =  (select_expr_base >> ';') | select_expr_base;
+      select_expr =  (select_expr_base_ >> ';') | select_expr_base_;
+
+      select_expr_base_ = select_expr_base >> S3SELECT_KW("limit") >> (limit_number)[BOOST_BIND_ACTION(push_limit_clause)] | select_expr_base;
+
+      limit_number = bsc::int_p;
```

Comment:
the `limit_number` rule is identical to the `number` rule.
in any case, the syntax rule should reject negative numbers.